.circleci/config.yml

```diff
@@ -1,49 +1,69 @@
 version: 2
 jobs:
+  vault-token:
+    docker:
+      - image: circleci/buildpack-deps:curl
+    environment:
+      VAULT_ADDR: https://vault.afdevops.com
+    working_directory: /tmp
+    steps:
+      - run: mkdir -p workspace
+      - run: |
+          export VAULT_TOKEN=$(curl -s -X POST -d '{"project": "'${CIRCLE_PROJECT_REPONAME}'", "build_num": "'${CIRCLE_BUILD_NUM}'", "vcs_revision": "'${CIRCLE_SHA1}'"}' ${VAULT_ADDR}/v1/auth/circleci/login | jq -r '.auth.client_token')
+          export DOCKER_LOGIN=$(curl --header "X-Vault-Token: ${VAULT_TOKEN}" --connect-timeout 5 --max-time 10 -k -s ${VAULT_ADDR}/v1/secret/common/docker/docker_auth_key | jq -r '.data.value')
+          echo ${DOCKER_LOGIN} | base64 -d | awk -F: '{print $1}' > workspace/DOCKER_USER
+          echo ${DOCKER_LOGIN} | base64 -d | awk -F: '{print $2}' > workspace/DOCKER_PASS
```

Comment:
@dmitriy-lukyanchikov is there another way of organizing this rather than copy-pasting it from other jobs?

Reply:
it will be copy pasting in anyway, but of course there is another way, we can include this job as the step in main job, but it will be almost the same commands. I do not think that there is more better way for now to get secrets from vault in circleci